gcloud-java-dns/src/main/java/com/google/gcloud/dns/DnsImpl.java

```diff
@@ -309,11 +315,86 @@ public ChangeRequest getChangeRequest(final String zoneName, final String change
     }
   }
 
+  @Override
+  public void submitBatch(DnsBatch toSubmit) {
+    try {
+      BatchRequest batchRequest = prepareBatch(toSubmit);
+      batchRequest.execute();
+    } catch (IOException ex) {
+      throw new DnsException(ex);
+    }
+  }
+
+  /**
+   * Since {@code BatchRequest} is a final class, it cannot be mocked with easy mock and the call of
+   * {@code execute()} cannot be tested. Thus, most of the functionality of {@link
+   * #submitBatch(DnsBatch)} is extracted to this method which does not make the call so it does not
+   * communicate with the service.
+   */
+  BatchRequest prepareBatch(DnsBatch toSubmit) throws IOException {
+    BatchRequest batch = null;
+    for (Map.Entry<DnsBatch.Request, DnsBatch.Callback> entry : toSubmit.requests().entrySet()) {
+      DnsBatch.Request request = entry.getKey();
+      DnsBatch.Callback callback = entry.getValue();
+      switch (request.operation()) {
+        case LIST_ZONES:
+          JsonBatchCallback rpcCallback = listZonesCallback(callback, request);
+          batch =
+              dnsRpc.enqueueListZones(batch, request, rpcCallback, optionMap(request.options()));
+          break;
+        default:
+          // todo(mderka) implement the rest of the operations
+          throw new UnsupportedOperationException("Not implemented yet");
+      }
+    }
+    return batch;
+  }
+
+  @Override
+  public DnsBatch batch() {
+    return new DnsBatch(this);
+  }
+
+  private JsonBatchCallback listZonesCallback(final DnsBatch.Callback callback,
+      final DnsBatch.Request request) {
+    return new JsonBatchCallback<ManagedZonesListResponse>() {
+      @Override
+      public void onFailure(GoogleJsonError error, HttpHeaders httpHeaders) {
+        if (callback != null) {
```

Comment:
I think that a `null` callback should be supported. The user may want to just create bunch of zones without caring about the result.